sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -720,7 +734,7 @@ intervalUnit
     ;
 
 colPosition
-    : FIRST | AFTER identifier
+    : FIRST | AFTER qualifiedName
```

Comment:
Do we need to modify this? I'm not sure what happens if something like:

```
ALTER TABLE tbl ADD a.b.c ... AFTER x.y
```

Reply:
Yes, this is needed because columns are identified by `qualifiedName`. The parser shouldn't fail if a nested column name is used. Instead, analysis should catch problems.

Reply:
I should also add that it is valid to reorder nested fields:

```
ALTER TABLE tbl ADD point.z bigint AFTER point.y
```